qiskit_nature/second_q/mappers/bosonic_logarithmic_mapper.py

```diff
@@ -0,0 +1,235 @@
+        Hamiltonian Construction, and Error Analysis, Arxiv https://doi.org/10.48550/arXiv.2307.06580
+
+        [2] R. Somma et al., Quantum Simulations of Physics Problems, Arxiv
+        https://doi.org/10.48550/arXiv.quant-ph/0304063
+    """
+
+    def __init__(self, max_occupation: int) -> None:
+        # Compute the actual max occupation from the one selected by the user
+        self.number_of_qubits_per_mode: int = (
+            1 if max_occupation == 0 else math.ceil(np.log2(max_occupation + 1))
+        )
+        max_calculated_occupation = 2**self.number_of_qubits_per_mode - 1
+        if max_occupation != max_calculated_occupation:
+            logger.warning(
+                f"The user requested a max occupation of {max_occupation}, but the actual "
+                + f"max occupation is {max_calculated_occupation}."
+            )
+        super().__init__(max_calculated_occupation)
+
+    @property
+    def number_of_qubits_per_mode(self) -> int:
+        """The minimum number of qubits required to represent a bosonic mode given a max occupation."""
+        return self._number_of_qubits_per_mode
+
+    @number_of_qubits_per_mode.setter
+    def number_of_qubits_per_mode(self, num_qubits: int) -> None:
+        if num_qubits < 1:
+            raise ValueError(f"The number of qubits must be at least 1, and not {num_qubits}.")
+        self._number_of_qubits_per_mode: int = num_qubits
+
+    def _map_single(
+        self, second_q_op: BosonicOp, *, register_length: int | None = None
+    ) -> SparsePauliOp:
+        """Maps a :class:`~qiskit_nature.second_q.operators.SparseLabelOp` to a ``SparsePauliOp``.
+
+        Args:
+            second_q_op: the ``SparseLabelOp`` to be mapped.
+            register_length: when provided, this will be used to overwrite the ``register_length``
+                attribute of the operator being mapped. This is possible because the
+                ``register_length`` is considered a lower bound in a ``SparseLabelOp``.
+
+        Returns:
+            The qubit operator corresponding to the problem-Hamiltonian in the qubit space.
+        """
+        if register_length is None:
+            register_length = second_q_op.num_modes
+
+        # The actual register length is the number of qubits per mode times the number of modes
+        qubit_register_length: int = register_length * self.number_of_qubits_per_mode
+        # Create a Pauli operator, which we will fill in this method
+        pauli_op: list[SparsePauliOp] = []
+        # Then we loop over all the terms of the bosonic operator
+        for terms, coeff in second_q_op.terms():
+            # Then loop over each term (terms -> List[Tuple[string, int]])
+            bos_op_to_pauli_op = SparsePauliOp(["I" * qubit_register_length], coeffs=[1.0])
+            # Loop over the operators in the term
+            for op, idx in terms:
+                if op not in ("+", "-"):
+                    break
```

Comment:
Just seeing this now. I also see that this is done in `BosonicLinearMapper`. I don't think this is a good idea because it lets the other for-loops continue. Instead, we should raise a `ValueError` with some helpful message indicating that an invalid operator type was found.

Please make the same change in the linear mapper.

Reply:
Yeah that's a good idea. It does not make sense to map an invalid operator